drivers/led/lp50xx.c

```diff
@@ -0,0 +1,341 @@
+/*
+ * Copyright (c) 2020 Seagate Technology LLC
+ * Copyright (c) 2022 Grinn
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * @brief LP50xx LED controller
+ */
+#include <errno.h>
+#include <zephyr/device.h>
+#include <zephyr/devicetree.h>
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/drivers/led.h>
+#include <zephyr/drivers/led/lp50xx.h>
+#include <zephyr/kernel.h>
+#include <zephyr/sys/util.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(lp50xx, CONFIG_LED_LOG_LEVEL);
+
+#define LP50XX_MIN_BRIGHTNESS		0U
+#define LP50XX_MAX_BRIGHTNESS		100U
+
+#define LP5012_NUM_MODULES	4
+#define LP5024_NUM_MODULES	8
+#define LP5036_NUM_MODULES	12
+
+#define LP50XX_MAX_CHANNELS(nmodules)	\
+	((LP50XX_COLORS_PER_LED + 1) * ((nmodules) + 1))
+
+/* Base registers */
+#define LP50XX_DEVICE_CONFIG0		0x00
+#define LP50XX_DEVICE_CONFIG1		0x01
+#define LP50XX_LED_CONFIG0		0x02
+
+#define LP50XX_BANK_BASE(chip)			\
+	((chip) < LP5030 ? 0x03 : 0x04)
+
+#define LP50XX_LED0_BRIGHTNESS(base)		\
+	((base) + 4)
+
+#define LP50XX_OUT0_COLOR(base, nmodules)	\
+	(LP50XX_LED0_BRIGHTNESS(base) + (nmodules))
```

Comment:
Can't we deduce the bank base from `nmodules` ?

```
#define LP50XX_BANK_BASE(nmodules)      (3 + (nmodules - 1) / 8)

#define LP50XX_LED0_BRIGHTNESS(nmodules)        \
        (LP50XX_BANK_BASE(nmodules) + 4)
        
#define LP50XX_OUT0_COLOR(nmodules)             \
        (LP50XX_LED0_BRIGHTNESS(nmodules) + (nmodules))

#define LP50XX_RESET(nmodules)                  \
        (LP50XX_OUT0_COLOR(nmodules) + LP50XX_COLORS_PER_LED * (nmodules))
```

Reply:
Well spotted